packages/material-ui/src/utils/reactHelpers.test.js

```diff
@@ -83,6 +92,91 @@ describe('utils/reactHelpers.js', () => {
       mount(<Component innerRef={outerRef} />);
 
       assert.strictEqual(outerRef.current.textContent, 'has a ref');
+      assert.strictEqual(consoleErrorMock.callCount(), 0);
+    });
+
+    it('forks if only one of the branches requires a ref', () => {
+      const Component = React.forwardRef(function Component(props, ref) {
+        const [hasRef, setHasRef] = React.useState(false);
+        const handleOwnRef = React.useCallback(() => setHasRef(true), []);
+        const handleRef = useForkRef(handleOwnRef, ref);
+
+        return <div ref={handleRef}>{String(hasRef)}</div>;
+      });
+
+      const wrapper = mount(<Component />);
+
+      assert.strictEqual(wrapper.containsMatchingElement(<div>true</div>), true);
+      assert.strictEqual(consoleErrorMock.callCount(), 0);
+    });
+
+    it('does nothing if none of the forked branches requires a ref', () => {
+      const Outer = React.forwardRef(function Outer(props, ref) {
+        const { children } = props;
+        const handleRef = useForkRef(children.ref, ref);
+
+        return React.cloneElement(children, { ref: handleRef });
+      });
+
+      Outer.propTypes = { children: PropTypes.element.isRequired };
+
+      function Inner() {
+        return <div />;
+      }
+
+      mount(
+        <Outer>
+          <Inner />
+        </Outer>,
+      );
+      assert.strictEqual(consoleErrorMock.callCount(), 0);
+    });
+
+    describe('changing refs', () => {
+      // use named props rather than ref attribute because enzyme ignores
+      // ref attributes on the root component
+      function Div(props) {
+        const { leftRef, rightRef, ...other } = props;
+        const handleRef = useForkRef(leftRef, rightRef);
+
+        return <div {...other} ref={handleRef} />;
+      }
+
+      Div.propTypes = {
+        leftRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
+        rightRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
+      };
+
+      it('handles changing from no ref to some ref', () => {
+        const wrapper = mount(<Div id="test" />);
+
+        assert.strictEqual(consoleErrorMock.callCount(), 0);
+
+        const ref = React.createRef();
+        wrapper.setProps({ leftRef: ref });
+
+        assert.strictEqual(ref.current.id, 'test');
+        assert.strictEqual(consoleErrorMock.callCount(), 0);
+      });
+
+      it('cleans up detached refs', () => {
+        const firstLeftRef = React.createRef();
+        const firstRightRef = { current: null, _id: 'first-inner' };
+        const secondRightRef = { current: null, _id: 'second-inner' };
```

Comment:
```suggestion
        const secondRightRef = React.createRef();
```

Reply:
I assumed it was done this way to improve the debugging experience of the test. It looks good either way :)